# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build cairo-build harfbuzz-old-build"

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.63; am-1.10; lt-1.5.24)
libtoolize -c -f; aclocal -I . -I m4; autoheader; autoconf; automake -a -c
rm -rf pango/opentype
PKG_CONFIG_PATH=/dfb/lib/pkgconfig ./configure --with-system-harfbuzz --prefix=/dfb

# build
make

# install
install -d $DESTDIR/dfb/bin
install pango/.libs/pango-querymodules $DESTDIR/dfb/bin
install pango-view/.libs/pangoft2-view $DESTDIR/dfb/bin/pango-view
install -d $DESTDIR/dfb/etc/pango
install -d $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-attributes.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-bidi-type.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-break.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-context.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-coverage.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-engine.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-enum-types.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-features.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-font.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-fontmap.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-fontset.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-glyph-item.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-glyph.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-gravity.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-item.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-language.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-layout.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-matrix.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-renderer.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-script.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-tabs.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-types.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pango-utils.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pangocairo.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pangofc-decoder.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pangofc-font.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pangofc-fontmap.h $DESTDIR/dfb/include/pango-1.0/pango
install -m 644 pango/pangoft2.h $DESTDIR/dfb/include/pango-1.0/pango
install -d $DESTDIR/dfb/lib
install pango/.libs/libpango-1.0.so.0 $DESTDIR/dfb/lib
ln -sf libpango-1.0.so.0 $DESTDIR/dfb/lib/libpango-1.0.so
install pango/.libs/libpangocairo-1.0.so.0 $DESTDIR/dfb/lib
ln -sf libpangocairo-1.0.so.0 $DESTDIR/dfb/lib/libpangocairo-1.0.so
install pango/.libs/libpangoft2-1.0.so.0 $DESTDIR/dfb/lib
ln -sf libpangoft2-1.0.so.0 $DESTDIR/dfb/lib/libpangoft2-1.0.so
install -d $DESTDIR/dfb/lib/pango/1.6.0/modules
install modules/basic/.libs/pango-basic-fc.so $DESTDIR/dfb/lib/pango/1.6.0/modules
install -d $DESTDIR/dfb/lib/pkgconfig
install -m 644 pango.pc $DESTDIR/dfb/lib/pkgconfig
install -m 644 pangocairo.pc $DESTDIR/dfb/lib/pkgconfig
install -m 644 pangoft2.pc $DESTDIR/dfb/lib/pkgconfig
LD_LIBRARY_PATH=$DESTDIR/dfb/lib $DESTDIR/dfb/bin/pango-querymodules > $DESTDIR/dfb/etc/pango/pango.modules

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/pango-build.pc << EOF
Name: pango
Version: 1.22.4
Description: A library for layout and rendering of text
Requires: $REQUIRES

devel=\\
/dfb/include/pango-1.0/pango/pango.h \\
/dfb/include/pango-1.0/pango/pango-attributes.h \\
/dfb/include/pango-1.0/pango/pango-bidi-type.h \\
/dfb/include/pango-1.0/pango/pango-break.h \\
/dfb/include/pango-1.0/pango/pango-context.h \\
/dfb/include/pango-1.0/pango/pango-coverage.h \\
/dfb/include/pango-1.0/pango/pango-engine.h \\
/dfb/include/pango-1.0/pango/pango-enum-types.h \\
/dfb/include/pango-1.0/pango/pango-features.h \\
/dfb/include/pango-1.0/pango/pango-font.h \\
/dfb/include/pango-1.0/pango/pango-fontmap.h \\
/dfb/include/pango-1.0/pango/pango-fontset.h \\
/dfb/include/pango-1.0/pango/pango-glyph-item.h \\
/dfb/include/pango-1.0/pango/pango-glyph.h \\
/dfb/include/pango-1.0/pango/pango-gravity.h \\
/dfb/include/pango-1.0/pango/pango-item.h \\
/dfb/include/pango-1.0/pango/pango-language.h \\
/dfb/include/pango-1.0/pango/pango-layout.h \\
/dfb/include/pango-1.0/pango/pango-matrix.h \\
/dfb/include/pango-1.0/pango/pango-renderer.h \\
/dfb/include/pango-1.0/pango/pango-script.h \\
/dfb/include/pango-1.0/pango/pango-tabs.h \\
/dfb/include/pango-1.0/pango/pango-types.h \\
/dfb/include/pango-1.0/pango/pango-utils.h \\
/dfb/include/pango-1.0/pango/pangocairo.h \\
/dfb/include/pango-1.0/pango/pangofc-decoder.h \\
/dfb/include/pango-1.0/pango/pangofc-font.h \\
/dfb/include/pango-1.0/pango/pangofc-fontmap.h \\
/dfb/include/pango-1.0/pango/pangoft2.h \\
/dfb/lib/libpango-1.0.so \\
/dfb/lib/libpangocairo-1.0.so \\
/dfb/lib/libpangoft2-1.0.so \\
/dfb/lib/pkgconfig/pango.pc \\
/dfb/lib/pkgconfig/pangocairo.pc \\
/dfb/lib/pkgconfig/pangoft2.pc

exec=\\
/dfb/bin/pango-querymodules \\
/dfb/bin/pango-view \\
/dfb/etc/pango/pango.modules \\
/dfb/lib/libpango-1.0.so.0 \\
/dfb/lib/libpangocairo-1.0.so.0 \\
/dfb/lib/libpangoft2-1.0.so.0 \\
/dfb/lib/pango/1.6.0/modules/pango-basic-fc.so
EOF

exit

# patch
--- pango-1.22.4.orig/configure.in
+++ pango-1.22.4/configure.in
@@ -47,6 +47,7 @@
 AM_INIT_AUTOMAKE(1.9 gnits dist-bzip2 no-dist-gzip)
 AM_CONFIG_HEADER([config.h])
 
+AM_MAINTAINER_MODE
 
 PANGO_VERSION_MAJOR=pango_version_major()
 PANGO_VERSION_MINOR=pango_version_minor()
@@ -198,6 +199,10 @@
                               [disable all source autogeneration rules])],,
               [enable_rebuilds=yes])
 
+AC_ARG_WITH(system-harfbuzz,
+   AC_HELP_STRING([--with-system-harfbuzz],
+                  [use system harfbuzz library @<:@default=no@:>@]))
+
 AM_CONDITIONAL(CROSS_COMPILING, [test $cross_compiling = yes])
 
 # define a MAINT-like variable REBUILD which is set if Perl
@@ -250,7 +255,17 @@
   # By freetype, we really always mean freetype+fontconfig
   FREETYPE_CFLAGS="$FREETYPE_CFLAGS $FONTCONFIG_CFLAGS"
   FREETYPE_LIBS="$FREETYPE_LIBS $FONTCONFIG_LIBS"
+
+  if test "$with_system_harfbuzz" = "yes"; then
+    PKG_CHECK_MODULES(OPENTYPE, harfbuzz-old)
+  else
+    OPENTYPE_LIBS="opentype/libharfbuzz-1.la"
+    OPENTYPE_DEPS="$OPENTYPE_LIBS"
+    OPENTYPE_SUBDIR=opentype
+  fi
 fi
+AC_SUBST(OPENTYPE_DEPS)
+AC_SUBST(OPENTYPE_SUBDIR)
 AM_CONDITIONAL(HAVE_FREETYPE, $have_freetype)
 
 #
@@ -938,7 +953,6 @@
 Makefile 
 pango/Makefile
 pango/mini-fribidi/Makefile
-pango/opentype/Makefile
 pango/pango.rc
 pango/pangoft2.rc
 pango/pangowin32.rc
@@ -973,6 +987,10 @@
 pango-zip.sh
 ])
 
+if test "$with_system_harfbuzz" != "yes"; then
+  AC_CONFIG_FILES([pango/opentype/Makefile])
+fi
+
 AC_CONFIG_FILES([tests/runtests.sh],
 	        [chmod +x tests/runtests.sh])
 
--- pango-1.22.4.orig/m4/gtk-doc.m4
+++ pango-1.22.4/m4/gtk-doc.m4
@@ -0,0 +1,39 @@
+dnl -*- mode: autoconf -*-
+
+# serial 1
+
+dnl Usage:
+dnl   GTK_DOC_CHECK([minimum-gtk-doc-version])
+AC_DEFUN([GTK_DOC_CHECK],
+[
+  AC_BEFORE([AC_PROG_LIBTOOL],[$0])dnl setup libtool first
+  AC_BEFORE([AM_PROG_LIBTOOL],[$0])dnl setup libtool first
+  dnl for overriding the documentation installation directory
+  AC_ARG_WITH([html-dir],
+    AS_HELP_STRING([--with-html-dir=PATH], [path to installed docs]),,
+    [with_html_dir='${datadir}/gtk-doc/html'])
+  HTML_DIR="$with_html_dir"
+  AC_SUBST([HTML_DIR])
+
+  dnl enable/disable documentation building
+  AC_ARG_ENABLE([gtk-doc],
+    AS_HELP_STRING([--enable-gtk-doc],
+                   [use gtk-doc to build documentation [[default=no]]]),,
+    [enable_gtk_doc=no])
+
+  if test x$enable_gtk_doc = xyes; then
+    ifelse([$1],[],
+      [PKG_CHECK_EXISTS([gtk-doc],,
+                        AC_MSG_ERROR([gtk-doc not installed and --enable-gtk-doc requested]))],
+      [PKG_CHECK_EXISTS([gtk-doc >= $1],,
+                        AC_MSG_ERROR([You need to have gtk-doc >= $1 installed to build gtk-doc]))])
+  fi
+
+  AC_MSG_CHECKING([whether to build gtk-doc documentation])
+  AC_MSG_RESULT($enable_gtk_doc)
+
+  AC_PATH_PROGS(GTKDOC_CHECK,gtkdoc-check,)
+
+  AM_CONDITIONAL([ENABLE_GTK_DOC], [test x$enable_gtk_doc = xyes])
+  AM_CONDITIONAL([GTK_DOC_USE_LIBTOOL], [test -n "$LIBTOOL"])
+])
--- pango-1.22.4.orig/pango/Makefile.am
+++ pango-1.22.4/pango/Makefile.am
@@ -4,15 +4,11 @@
 # which are more like other make's VPATH.
 GPATH = $(srcdir)
 
-if HAVE_FREETYPE
-OPENTYPE_SUBDIR=opentype
-endif
-
 FRIBIDI_SUBDIR=mini-fribidi
 
 SUBDIRS = $(OPENTYPE_SUBDIR) $(FRIBIDI_SUBDIR)
 
-DIST_SUBDIRS = mini-fribidi opentype
+DIST_SUBDIRS = mini-fribidi
 
 INCLUDES =						\
 	-DG_LOG_DOMAIN=\"Pango\"			\
@@ -191,16 +187,17 @@
 lib_LTLIBRARIES += libpangoft2-1.0.la
 endif
 
+libpangoft2_1_0_la_CFLAGS = $(OPENTYPE_CFLAGS)
 libpangoft2_1_0_la_LDFLAGS = $(LIBRARY_LIBTOOL_OPTIONS)
 libpangoft2_1_0_la_LIBADD =			\
-	opentype/libharfbuzz-1.la		\
+	$(OPENTYPE_LIBS)			\
 	libpango-$(PANGO_API_VERSION).la	\
 	$(INCLUDED_FC_MODULES)			\
 	$(GLIB_LIBS)				\
 	$(FREETYPE_LIBS)			\
 	$(libm)
 libpangoft2_1_0_la_DEPENDENCIES =		\
-	opentype/libharfbuzz-1.la		\
+	$(OPENTYPE_DEPS)			\
 	libpango-$(PANGO_API_VERSION).la	\
 	$(INCLUDED_FC_MODULES)
 libpangoft2_1_0_la_SOURCES =   	\

# source
https://download.gnome.org/sources/pango/1.22/pango-1.22.4.tar.gz
